lib/src/testing/test_file.dart

```diff
@@ -169,6 +169,17 @@ class FormatTest {
   }
 }
 
+extension SourceCodeExtensions on SourceCode {
```

Comment:
Why does this live in lib/src/testing instead of the example directly?

Reply:
I wanted it to be next to the code in test_file.dart that parses the selection markers out of the string. Also, at some point, it wouldn't hurt if the test code used this to produce better error messages when tests fail because the selection is wrong.